playbooks/common/openshift-cluster/initialize_facts.yml

```diff
@@ -6,12 +6,140 @@
 
 - name: Initialize host facts
   hosts: oo_all_hosts
-  roles:
-  - openshift_facts
   tasks:
-  - openshift_facts:
+  - name: load openshift_facts module
+    include_role:
+      name: openshift_facts
+
+  # TODO: Should this role be refactored into health_checks??
+  - name: Run openshift_sanitize_inventory to set variables
+    include_role:
```

Comment:
@ingvagabund, heh, I agree that at first the new style appears different but I can assure you the benefits are there.  We discussed this very topic with James from Ansible and he was telling us that when you use `include_role` the playbook sections `pre_tasks`, `roles`, `tasks`, and `post_tasks` ordering goes away.  The flow of the playbook moves top to bottom and the flexibility increases as you can include only tasks from a role that you desire.  With `roles:` you get the entire role if that is desired or not.

I can find the meeting recording and send it to you.  I think its in the architecture notes.